dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -34,8 +35,27 @@ async def snapshot(self, req):
             "ray_version": ray.__version__,
             "ray_commit": ray.__commit__
         }
+
+        # Serve deployment names are used as keys but should not be converted
+        # to camel case because different user-provided names may map to the
+        # same name in camel case, e.g. "my_func" and "myFunc". For all other
+        # keys, convert to camel case as usual for backward compatibility.
+        converted_case_snapshot = dict()
+        for name, data in snapshot.items():
+            if name == "deployments":
+                converted_case_snapshot["deployments"] = dict()
+                for deployment_name, info in data.items():
+                    converted_case_snapshot["deployments"][
+                        deployment_name] = to_google_style(info)
+            else:
+                converted_case_snapshot[to_camel_case(name)] = to_google_style(
+                    data)
+
         return dashboard_utils.rest_response(
-            success=True, message="hello", snapshot=snapshot)
+            success=True,
+            message="hello",
+            convert_case=False,
+            snapshot=converted_case_snapshot)
```

Comment:
oh man, yeah, this looks pretty messy... I'm fine w/ the hashing solution and considering un-camel-casing everything in the future